cpmpy/tools/mus.py

```diff
@@ -29,8 +31,11 @@ def mus(soft, hard=[], solver="ortools"):
     """
     # order so that constraints with many variables are tried and removed first
     candidates = sorted(soft, key=lambda c: -len(get_variables(c)))
-    
+
     assump = boolvar(shape=len(soft), name="assump")
+    if len(soft) == 1:
```

Comment:
I think this should actually be resolved in `core.py`.
What we want is equivalent behaviour to: `bv.implies(all([some_cons]))` which extracts the 1 element from the list on the rhs and returns a single expression.
So maybe we can check wether the rhs is a list when calling `.implies()` in an expression that is not a NDVarArray?
Anyway, if that's too much hassle this fix is fine for me :)

Reply:
if you leave this, maybe put a comment an refer to this bug request?

Reply:
That would be a small improvement in core.py especially for the case where the implied constraint is either a single boolvar or an empty list.
For instance
```python
from cpmpy import *; 
import cpmpy as cp; 
bv, bv2 = boolvar(shape=2);
some_cons = [~bv2];
```

```python
>>> bv.implies(some_cons2)
(BV0) -> [~BV1]
>>>bv.implies([])
(BV0) -> []
```